### ISSUE TITLE
[Menu] Menu component is missing the action trigger

### ISSUE THREAD

**Issue by alina-jacob** [carbon-design-system/carbon#17971]
## Acceptance criteria
- Menu is a primitive component that is invoked by an action trigger like a button or context menu. Our current implementation doesn't have a trigger that launches the menu, making it an incomplete representation of the intended interaction. 

<br>

**Current implementation**
The [Menu component](https://react.carbondesignsystem.com/?path=/story/components-menu--playground) is open by default and doesn't close at all. This is incorrect representation of the interaction. 

**Expected behavior**
The Menu component is usually triggered by an interactive element which could be any of the following: a button, menu button, combo-button, overflow menu or can have nothing like the context menu. 
The intent is that the first tab should land on "something" and then it should open the menu, because there are some interaction within the menu, which upon activation should close the menu. For instance, if you select/deselect a preference, the menu will collapse, as coded over here in the [v2 Overflow menu](https://react.carbondesignsystem.com/?path=/story/components-overflowmenu-feature-flag--nested). 

**Solution**
There are multiple ways this can be tackled. 

**_Simple option:_**
The simple option would be just updating the storybook documentation to inform users that an ideal implementation exists in the [v2 Overflow menu](https://react.carbondesignsystem.com/?path=/story/components-overflowmenu-feature-flag--nested) story. 

**_Mid option:_**
Create a new story or update the current that has the same menu, just with a trigger action. Inspired by a similar primitive component, the Popover ([normal](https://react.carbondesignsystem.com/?path=/story/components-popover--playground), [in-context](https://react.carbondesignsystem.com/?path=/story/components-popover--tab-tip))

**_Complex option:_**
Introduce a thorough Menu interaction for all components that have complex, selectable and nested menu-items in their stories. Current list includes Menu button, Combo button, Overflow menu and maybe UI shell header. There could be more.
